README.md

```diff
@@ -91,16 +219,627 @@ impl DeserializeFromValue<MyError> for Name {
+    #[deserr(skip)]
+    hidden: usize,
+}
+
+let data = deserialize::<Search, _, JsonError>(
+    json!({ "query": "doggo" }),
+)
+.unwrap();
+assert_eq!(data, Search { query: String::from("doggo"), hidden: 0 });
+
+// if you try to specify the field, it is ignored
+let data = deserialize::<Search, _, JsonError>(
+    json!({ "query": "doggo", "hidden": 2 }),
+)
+.unwrap();
+assert_eq!(data, Search { query: String::from("doggo"), hidden: 0 });
+
+// Here, we're going to see how skip interacts with `deny_unknown_fields`
+
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+#[deserr(deny_unknown_fields)]
+struct Search2 {
+    query: String,
+    // A field can be skipped if it implements `Default`.
+    #[deserr(skip)]
+    hidden: usize,
+}
+
+let error = deserialize::<Search2, _, JsonError>(
+    json!({ "query": "doggo", "hidden": 1 }),
+)
+.unwrap_err();
+// NOTE: `hidden` isn't in the list of expected fields + `hidden` is effectively considered as a non-existing field.
+assert_eq!(error.to_string(), "Json deserialize error: unknown field `hidden`, expected one of `query` at ``.");
+```
+
+#### `map`
+
+Map a field **after** it has been deserialized.
+
+```rust
+use deserr::{DeserializeFromValue, deserialize};
+use deserr::serde_json::JsonError;
+use serde_json::json;
+
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+struct Search {
+    query: String,
+    #[deserr(map = add_one)]
+    limit: usize,
 }
+
+fn add_one(n: usize) -> usize {
+    n.saturating_add(1)
+}
+
+let data = deserialize::<Search, _, JsonError>(
+    json!({ "query": "doggo", "limit": 0 }),
+)
+.unwrap();
+assert_eq!(data, Search { query: String::from("doggo"), limit: 1 });
+
+// Let's see how `map` interacts with the `default` attributes.
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+struct Search2 {
+    query: String,
+    #[deserr(default, map = add_one)]
+    limit: usize,
+}
+
+let data = deserialize::<Search2, _, JsonError>(
+    json!({ "query": "doggo" }),
+)
+.unwrap();
+// As we can see, the `map` attribute is applied AFTER the `default`.
+assert_eq!(data, Search2 { query: String::from("doggo"), limit: 1 });
 ```
 
+#### `missing_field_error`
+
+Gives you the opportunity to customize the error message if this specific field
+if missing.
+
+```rust
+use deserr::{DeserializeFromValue, DeserializeError, ValuePointerRef, ErrorKind, deserialize};
+use deserr::serde_json::JsonError;
+use serde_json::json;
+use std::convert::Infallible;
+
+#[derive(DeserializeFromValue, Debug, PartialEq, Eq)]
+struct Search {
+    #[deserr(missing_field_error = missing_query_field)]
+    query: String,
+    limit: usize,
+}
+
+fn missing_query_field<E: DeserializeError>(_field_name: &str, location: ValuePointerRef) -> E {
+    deserr::take_result_content(E::error::<Infallible>(
+        None,
+        ErrorKind::Unexpected {
+            msg: String::from("I really need the query field, please give it to me uwu"),
+        },
+        location,
+    ))
+}
```

Comment:
I would really like at least one example where a custom function provided by the user returns a non-generic type. Maybe we can add one *after* the `error` container/field attribute is explained. Maybe also as an example for the `MergeWithError` trait?

Reply:
Yep that's true, it would be nice to show how to use the `MergeWithError` trait